internal/error.go

```diff
@@ -689,46 +720,48 @@ func convertFailureToError(failure *failurepb.Failure, dc DataConverter) error {
 		return nil
 	}
 
+	var err error
+
 	if failure.GetApplicationFailureInfo() != nil {
 		applicationFailureInfo := failure.GetApplicationFailureInfo()
 		details := newEncodedValues(applicationFailureInfo.GetDetails(), dc)
 		switch applicationFailureInfo.GetType() {
 		case getErrorType(&ApplicationError{}):
-			return NewApplicationError(failure.GetMessage(), applicationFailureInfo.GetNonRetryable(), details)
+			err = NewApplicationError(failure.GetMessage(), applicationFailureInfo.GetNonRetryable(), details)
```

Comment:
Can user create an application error that contains any other error as a cause?

Reply:
No (`ApplicationError` does't have `cause`), but I can easily add this. Is there a requirement for this?

Reply:
For example It is possible to create application error because an activity failed.